engine/security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/DefaultOAuth2ServiceFactory.scala

```diff
@@ -19,36 +23,65 @@ class DefaultOAuth2Service(clientApi: OAuth2ClientApi[DefaultProfileResponse, De
-      val username = profile.login.getOrElse(profile.id.toString)
-      LoggedUser(id = profile.id.toString, username = username, rulesSet=rulesSet)
-    }
-  }
+  def authorize(token: String): Future[LoggedUser] = clientApi.profileRequest(token).map(getLoggedUser(_, configuration, allCategories))
 }
 
-class DefaultOAuth2ServiceFactory extends OAuth2ServiceFactory {
-  override def create(configuration: OAuth2Configuration, allCategories: List[String]): OAuth2Service =
-    DefaultOAuth2ServiceFactory.defaultService(configuration, allCategories)
-}
+class DefaultOAuth2ServiceFactoryWithProfileFormat[ProfileResponse: Decoder](implicit getLoggedUser: GetLoggedUserType[ProfileResponse]) {
+  def defaultService(configuration: OAuth2Configuration, allCategories: List[String]): OAuth2Service =
+    new DefaultOAuth2Service[ProfileResponse](OAuth2ClientApi[ProfileResponse, DefaultAccessTokenResponse](configuration), configuration, allCategories)
 
-object DefaultOAuth2ServiceFactory extends {
-  val defaultUserRole = "User"
+  def service(configuration: OAuth2Configuration, allCategories: List[String])(implicit backend: SttpBackend[Future, Nothing, NothingT], ec: ExecutionContext): OAuth2Service =
+    new DefaultOAuth2Service[ProfileResponse](new OAuth2ClientApi[ProfileResponse, DefaultAccessTokenResponse](configuration), configuration, allCategories)
+}
 
-  def apply(): DefaultOAuth2ServiceFactory = new DefaultOAuth2ServiceFactory()
+object DefaultOAuth2ServiceFactoryWithProfileFormat {
+  type GetLoggedUserType[ProfileResponse] = (ProfileResponse, OAuth2Configuration, List[String]) => LoggedUser
 
-  def defaultService(configuration: OAuth2Configuration, allCategories: List[String]): DefaultOAuth2Service =
-    new DefaultOAuth2Service(OAuth2ClientApi[DefaultProfileResponse, DefaultAccessTokenResponse](configuration), configuration, allCategories)
+  val defaultUserRole = "User"
 
-  def service(configuration: OAuth2Configuration, allCategories: List[String])(implicit backend: SttpBackend[Future, Nothing, NothingT]): DefaultOAuth2Service =
-    new DefaultOAuth2Service(new OAuth2ClientApi[DefaultProfileResponse, DefaultAccessTokenResponse](configuration), configuration, allCategories)
+  def apply(configuration: OAuth2Configuration, allCategories: List[String]) = {
+    import implicits._
+    configuration.profileFormat match {
+      case ProfileFormat.GITHUB => new DefaultOAuth2ServiceFactoryWithProfileFormat[GitHubProfileResponse]()
+      case ProfileFormat.AUTH0 => new DefaultOAuth2ServiceFactoryWithProfileFormat[Auth0ProfileResponse]()
+    }
+  }
 
   def getUserRoles(email: Option[String], configuration: OAuth2Configuration, defaults: List[String] = List(defaultUserRole)): List[String] =
     configuration
       .users
       .find(us => email.exists(_.toLowerCase.equals(us.identity.toLowerCase)))
       .map(_.roles ++ defaults)
       .getOrElse(defaults)
-}
\ No newline at end of file
+
+  object implicits {
```

Comment:
Let's put implementations, together with Auth0ProfileResponse and GithubProfileResponse 
in seperate file/package - so that it's more obvious what is needed